### ISSUE TITLE
Whiteout issue between btrfs and ext4

### ISSUE THREAD

**Issue by kenny1986** [docker/docker#17357]
I found this when using the `php:5-apache` official image. When building an image on btrfs, the image contains an extra `index.html` file in the `/var/www/html/` directory, while there is no such file on a normal ubuntu host with ext4

On a ubuntu host with btrfs:

```
ubuntu@ip-172-31-31-207:/var/lib/docker$ cat /etc/default/docker
# Docker Upstart and SysVinit configuration file
# Customize location of Docker binary (especially for development testing).
#DOCKER="/usr/local/bin/docker"
# Use DOCKER_OPTS to modify the daemon startup options.
#DOCKER_OPTS="-s btrfs --dns 8.8.8.8 --dns 8.8.4.4"
# If you need Docker to use an HTTP proxy, it can also be specified here.
#export http_proxy="http://127.0.0.1:3128/"
# This is also a handy place to tweak where Docker's temporary files go.
#export TMPDIR="/mnt/bigdrive/docker-tmp"
ubuntu@ip-172-31-31-207:/var/lib/docker$ df -T
Filesystem     Type     1K-blocks   Used Available Use% Mounted on
/dev/xvda1     ext4       8115168 855152   6824740  12% /
none           tmpfs            4      0         4   0% /sys/fs/cgroup
udev           devtmpfs    503188     12    503176   1% /dev
tmpfs          tmpfs       101632    340    101292   1% /run
none           tmpfs         5120      0      5120   0% /run/lock
none           tmpfs       508144      0    508144   0% /run/shm
none           tmpfs       102400      0    102400   0% /run/user
/dev/xvdb      btrfs      8388608 575040   7058528   8% /var/lib/docker
ubuntu@ip-172-31-31-207:/var/lib/docker$ docker pull php:5-apache
5-apache: Pulling from library/php
7a42f1433a16: Already exists 
3d88cbf54477: Already exists 
34edc74878ac: Already exists 
4e3cc2f6d768: Already exists 
022ec6897309: Already exists 
fb0c542d0e8d: Already exists 
76af60700c50: Already exists 
3bb902950e8c: Already exists 
7291266aac96: Already exists 
7439976e699e: Already exists 
be4d0249c6b2: Already exists 
4468017d46fa: Already exists 
04383a2eed1e: Already exists 
a8a5e32d28de: Already exists 
d54d520e7edc: Already exists 
a0a83488b6c3: Already exists 
69986cdd6a31: Already exists 
709f63e6ba7b: Already exists 
f7349be4b93d: Already exists 
d1353bf6e2ba: Already exists 
7ab917c221ac: Already exists 
3faa65935529: Already exists 
Digest: sha256:be76d2c4dfa6f79ffbaebbfb47f3ec7e60fd29f83e692e4b3fe76ef852168c23
Status: Image is up to date for php:5-apache
ubuntu@ip-172-31-31-207:/var/lib/docker$ docker run php:5-apache ls -la
total 12
drwxr-xr-x 1 www-data www-data    20 Oct 21 00:31 .
drwxr-xr-x 1 root     root         8 Oct 21 00:31 ..
-rw-r--r-- 1 root     root     11104 Oct 21 00:31 index.html
ubuntu@ip-172-31-31-207:/var/lib/docker$ docker version
Client:
 Version:      1.8.3
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   f4bf5c7
 Built:        Mon Oct 12 05:37:18 UTC 2015
 OS/Arch:      linux/amd64
Server:
 Version:      1.8.3
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   f4bf5c7
 Built:        Mon Oct 12 05:37:18 UTC 2015
 OS/Arch:      linux/amd64
ubuntu@ip-172-31-31-207:/var/lib/docker$ 
```

On a ubuntu host with ext4:
```
vagrant@vagrant-ubuntu-trusty-64:~$ cat /etc/default/docker
# Docker Upstart and SysVinit configuration file
# Customize location of Docker binary (especially for development testing).
#DOCKER="/usr/local/bin/docker"
# Use DOCKER_OPTS to modify the daemon startup options.
#DOCKER_OPTS="--dns 8.8.8.8 --dns 8.8.4.4"
# If you need Docker to use an HTTP proxy, it can also be specified here.
#export http_proxy="http://127.0.0.1:3128/"
# This is also a handy place to tweak where Docker's temporary files go.
#export TMPDIR="/mnt/bigdrive/docker-tmp"
vagrant@vagrant-ubuntu-trusty-64:~$ df -T
Filesystem     Type     1K-blocks    Used Available Use% Mounted on
/dev/sda1      ext4      41251136 9874404  29640820  25% /
none           tmpfs            4       0         4   0% /sys/fs/cgroup
udev           devtmpfs    237356      12    237344   1% /dev
tmpfs          tmpfs        50176     336     49840   1% /run
none           tmpfs         5120       0      5120   0% /run/lock
none           tmpfs       250868       0    250868   0% /run/shm
none           tmpfs       102400       0    102400   0% /run/user
vagrant@vagrant-ubuntu-trusty-64:~$ docker pull php:5-apache
5-apache: Pulling from library/php
e2a4fb18da48: Already exists 
58016a5acc80: Already exists 
927bb1e1227e: Already exists 
57f1a83d5202: Already exists 
e2f53f6703fb: Already exists 
d64f7cf5254c: Already exists 
111bb74b535b: Already exists 
189346d9f444: Already exists 
283a82a1ed7b: Already exists 
f430e74d68ff: Already exists 
11e215a8b8f2: Already exists 
f0ec0ad99787: Already exists 
74108b8f486e: Already exists 
8c31b4985b7b: Already exists 
723cc6be15f3: Already exists 
01203a3c7069: Already exists 
458b965428fe: Already exists 
50be39ed5ba9: Already exists 
3f926f31509c: Already exists 
8d5fa0e7e8a6: Already exists 
db71b861bdbd: Already exists 
0448546e1b12: Already exists 
Digest: sha256:be76d2c4dfa6f79ffbaebbfb47f3ec7e60fd29f83e692e4b3fe76ef852168c23
Status: Image is up to date for php:5-apache
vagrant@vagrant-ubuntu-trusty-64:~$ docker run php:5-apache ls -la
total 8
drwxr-xr-x 2 www-data www-data 4096 Oct 21 00:31 .
drwxr-xr-x 4 root     root     4096 Oct 21 00:31 ..
vagrant@vagrant-ubuntu-trusty-64:~$ docker version
Client:
 Version:      1.8.2
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   0a8c2e3
 Built:        Thu Sep 10 19:19:00 UTC 2015
 OS/Arch:      linux/amd64
Server:
 Version:      1.8.2
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   0a8c2e3
 Built:        Thu Sep 10 19:19:00 UTC 2015
 OS/Arch:      linux/amd64
vagrant@vagrant-ubuntu-trusty-64:~$ 
```

**Comment by tonistiigi** [comment]
This should be fixed with #16470 - that image contains `.opq` file that wasn't handled by btrfs.

**Comment by thaJeztah** [comment]
Thanks Toniis, I'll tentatively close this issue, but feel free to ping me if this is not resolved yet.

Thanks for reporting!
